### PR TITLE
Code reformatting with black

### PR DESCRIPTION
**Pull request by JAlvarezJarreta**
On most files, the difference is negligible, but I can see a few where the difference in format is significant and the result, in my opinion, an improvement in code readability.

### CODE REVIEW

File: lib/python/ensembl/brc4/runnable/load_sequence_data.py [hunk 1]
```diff
@@ -789,13 +874,11 @@ def unversion_scaffolds(self, cs_rank, logs):
                 self.copy_sr_name_to_syn(cs, xdb, self.pjc(logs, "cp2syn", cs))
                 self.sr_name_unversion(cs, "seq_region", "name", self.pjc(logs, "unv_sr", cs))
 
-
     def coord_sys_order(self, cs_order_str):
         cs_order_lst = map(lambda x: x.strip(), cs_order_str.split(","))
-        return { e:i for i,e in enumerate(filter(lambda x: len(x)>0, cs_order_lst)) }
+        return {e: i for i, e in enumerate(filter(lambda x: len(x) > 0, cs_order_lst))}
```
Comment (vsitnik) [inline lib/python/ensembl/brc4/runnable/load_sequence_data.py:879]: prefer to have e:i and i,e together :)
not critical

Comment (JAlvarezJarreta) [reply lib/python/ensembl/brc4/runnable/load_sequence_data.py:879]: I'm happy to revert particular preferences, but bear in mind `black` will do this change every time it is run over this file :+1:

Comment (MatBarba) [reply lib/python/ensembl/brc4/runnable/load_sequence_data.py:879]: I prefer them separated :P
I'd also prefer to have some space around the { ... } but we can't have everything :D

Comment (vsitnik) [reply lib/python/ensembl/brc4/runnable/load_sequence_data.py:879]: :P

Comment (vsitnik) [reply lib/python/ensembl/brc4/runnable/load_sequence_data.py:879]: if seriously, I don't mind :)

File: scripts/brc4/gff-overlaps.py [hunk 2]
```diff
@@ -24,138 +24,166 @@
...
-
-    parser = argparse.ArgumentParser(description='Scan a GFF file to detect overlapping seqfeature objects (default = gene).')
-    parser.add_argument('--gff', help='path of gff file to scan for overlaps', required=True, nargs='?' )
-    parser.add_argument('--stats', help='run stats analysis of GFF file instead of parsing GFF file', action='store_true' )
-    parser.add_argument('--logfile', help='path of file for logging (default=log.overlaps)', nargs='?', default='log.overlaps' )
-    parser.add_argument('--output', help='path of file for output (default=overlaps)', required=True, nargs='?', default='overlaps' )
-    parser.add_argument('--filter', help='seqfeature type to use for overlap isolation (default=gene)', nargs='?', default='gene' )
+
+
+if __name__ == "__main__":
```
Comment (vsitnik) [inline scripts/brc4/gff-overlaps.py:150]: I'd rather have a separate main function called from here

Comment (vsitnik) [reply scripts/brc4/gff-overlaps.py:150]: not with this PR!

File: scripts/chunk_fasta.py [hunk 3]
```diff
@@ -26,67 +26,95 @@
...
+    parser = argparse.ArgumentParser()
+    # in files
+    parser.add_argument(
+        "fasta_dna",
+        metavar="input.fna[.gz]",
+        type=str,
+        help="fasta (possibly gzipped) file with dna sequences to split",
+    )
+    # out
+    parser.add_argument(
```
Comment (vsitnik) [inline scripts/chunk_fasta.py:38]: don't like spreading add_argument arguments to individual lines

File: scripts/gff_metaparser/genmetaconf/manifest.py [hunk 4]
```diff
@@ -24,70 +24,70 @@
 
 from os.path import dirname, join as pj
 
-class Manifest:
-  def __init__(self, mapping, ungzip = True, always_copy = True):
```
Comment (vsitnik) [inline scripts/gff_metaparser/genmetaconf/manifest.py:28]: don't get this diff

File: scripts/gff_metaparser/genmetaconf/metaconf.py [hunk 5]
```diff
@@ -26,274 +26,286 @@
...
+            if re.match(r"^\s*$", raw):
+                continue
+            tag, *rest = raw.split(maxsplit=1)
+            if rest:
+                out[tag.strip()].append(rest[0].rstrip())
+                self._order[tag.strip()] = len(self._order)  # use the last rank for multi keys
+        # adding raw file path
+        meta_file_raw = abspath(tsv.name)
+        self.tech_data["META_FILE_RAW"].append(meta_file_raw.rstrip())
+        self._order["META_FILE_RAW"] = len(self._order)  # use the last rank for multi keys
```
Comment (vsitnik) [inline scripts/gff_metaparser/genmetaconf/metaconf.py:62]: don't get this diff

File: scripts/gff_metaparser/genmetaconf/seqregionconf.py [hunk 6]
```diff
@@ -26,364 +26,388 @@
...
-    #  get data from meta
-    self.fill_info_from_tech(meta)
-    #  fill synonyms from report file
-    self.fill_info_from_asm_rep(asm_rep_file)
-    #  load from seq_region_genebank, infer names based on the loaded synonyms
-    self.fill_info_from_seq_region_raw(seq_region_genbank, add_missing = False, update_from_new = True, update_syns = False)
-    #  load from seq_region_syns, infer names based on the loaded synonyms
-    self.fill_info_from_seq_region_syns(seq_region_syns)
-
-  def dump(self, json_out: str) -> None:
```
Comment (vsitnik) [inline scripts/gff_metaparser/genmetaconf/seqregionconf.py:93]: don't get this diff

File: scripts/brc4/get_metadata_from_redmine.py [hunk 7]
```diff
@@ -513,45 +535,52 @@ def check_organism_abbrev(name):
...
-    parser.add_argument('--output_dir', type=str,
-                help='Output_dir')
+    parser = argparse.ArgumentParser(description="Retrieve metadata from Redmine")
+
+    parser.add_argument("--key", type=str, required=True, help="Redmine authentification key")
+    parser.add_argument("--output_dir", type=str, help="Output_dir")
     # Choice
-    parser.add_argument('--get', choices=['genomes', 'organism_abbrev', 'abbreviate'], required=True,
-                help='Get genomes, or set organism_abbrev field (use update_redmine for actually changing it)')
+    parser.add_argument(
```
Comment (vsitnik) [inline scripts/brc4/get_metadata_from_redmine.py:546]: argparse args style

File: scripts/gff_metaparser/genmetaconf/seqregionconf.py (hunk 6 above)
Comment (MatBarba) [reply scripts/gff_metaparser/genmetaconf/seqregionconf.py:93]: 2 spaces to 4 spaces tab? So all lines affected

File: scripts/gff_metaparser/gff_stats.py [hunk 8]
```diff
@@ -28,89 +28,136 @@
...
-                      help="input gff file (use '-' to read from STDIN)" )
-  args = parser.parse_args()
-  if args.dump_used_options:
-    print("used options:", file=sys.stderr)
-    for o, v in args.__dict__.items():
-      if v: print("  %s: %s" %(o, v), file=sys.stderr)
-  return args
+    parser = argparse.ArgumentParser()
+    # configs
+    parser.add_argument(
```
Comment (vsitnik) [inline scripts/gff_metaparser/gff_stats.py:33]: argparse args style

File: lib/python/ensembl/brc4/runnable/load_sequence_data.py (hunk 1 above)
Comment (vsitnik) [reply lib/python/ensembl/brc4/runnable/load_sequence_data.py:879]: @JAlvarezJarreta we need to tune `black` or not use it :)

File: scripts/gff_metaparser/gff_stats.py (hunk 8 above)
Comment (MatBarba) [reply scripts/gff_metaparser/gff_stats.py:33]: no need to comment every instance of this reformatting, we can discuss it globally

File: scripts/gff_metaparser/gffstruct/gff3walker.py [hunk 9]
```diff
@@ -27,178 +27,176 @@
...
+            _PHASE=phase,
+            _FULLTAG=fulltag,
+            _DEPTH=depth,
+            _ISLEAF=is_leaf,
+            _RULESDATA=None,
+        )
+
+        # norm
+        new_id = self.norm_and_update_id(feat, quals, context)
+
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/gff3walker.py:138]: don't get diff. diff issues

File: scripts/gff_metaparser/gffstruct/rules/alias.py [hunk 10]
```diff
@@ -14,63 +14,67 @@
 # limitations under the License.
 
 
-# 
+#
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/rules/alias.py:17]: that was a meaningful space! the whole comment made obsolete :(
(don't fix with this PR, please :)

File: scripts/gff_metaparser/gffstruct/rules/alias.py [hunk 11]
```diff
@@ -14,63 +14,67 @@
...
 import sys
 import re
 
 from .base import BaseRule
 
 from collections import defaultdict
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/rules/alias.py:24]: "broken diff"

File: scripts/gff_metaparser/gffstruct/rules/alias.py (hunk 10 above)
Comment (MatBarba) [reply scripts/gff_metaparser/gffstruct/rules/alias.py:17]: What is that supposed to mean?

File: scripts/gff_metaparser/gffstruct/rules/base.py [hunk 12]
```diff
@@ -23,127 +23,142 @@
 from collections import defaultdict
 from copy import deepcopy, copy
 
+
 def _RulesType():
-  return {
-    "const_match" : defaultdict(list),
-    "_regex_match_pre" : defaultdict(list),
-    "regex_match" : list(),
-  }
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/rules/base.py:31]: "broken diff"

File: scripts/gff_metaparser/gffstruct/rules/fix_action.py [hunk 13]
```diff
@@ -20,385 +20,396 @@
...
+        self._action = None
+        self._action = self.parse(self._raw)
+        if self._additions > 0 and self._exclusions > 0:
+            print("%s has add and exclude operations at the same time. skipping" % self, file=sys.stderr)
+            self._action = None
+        if not self._action:
+            self._additions = 0
+            self._exclusions = 0
+            self._copy_leaves = 0
+
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/rules/fix_action.py:45]: "broken diff"

File: scripts/gff_metaparser/gffstruct/rules/gff.py [hunk 14]
```diff
@@ -20,206 +20,210 @@
 import sys
 from collections import defaultdict
 
+
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/rules/gff.py:23]: "broken diff"

File: scripts/gff_metaparser/gffstruct/rules/ignore.py [hunk 15]
```diff
@@ -16,12 +16,12 @@
 
 from .base import BaseRule
 
-class IgnoreRule(BaseRule):
```
Comment (vsitnik) [inline scripts/gff_metaparser/gffstruct/rules/ignore.py:19]: "broken diff"

**Review by vsitnik — APPROVED** [review]
all good.
1) Argparse add_argument arguments could remain grouped, but this is not critical.
2) diff is not trying hard -- not crirtical

P.S. What does this thing know about STYLE?!!

File: scripts/gff_metaparser/gffstruct/rules/alias.py (hunk 10 above)
Comment (vsitnik) [reply scripts/gff_metaparser/gffstruct/rules/alias.py:17]: no idea

Comment (vsitnik) [reply scripts/gff_metaparser/gffstruct/rules/alias.py:17]: mb. "TODO: add some doc"

File: scripts/gff_metaparser/genmetaconf/seqregionconf.py (hunk 6 above)
Comment (vsitnik) [reply scripts/gff_metaparser/genmetaconf/seqregionconf.py:93]: it's more of a problem with git. everything is shifted. mb issue with display on my side

File: scripts/gff_metaparser/genmetaconf/manifest.py (hunk 4 above)
Comment (ens-LCampbell) [reply scripts/gff_metaparser/genmetaconf/manifest.py:28]: Seems to me to be reformatting the level of indentation more than anything else

**Review by ens-LCampbell — APPROVED** [review]
Seems to be handling reformatting ok. I can't see anything egregious or broken in terms of unexpected code changes

**Comment by JAlvarezJarreta** [comment]
Since `black` is following PEP8, I'm afraid there is not much room for configuration of formatting, only these three options:
```
-l, --line-length INTEGER           How many characters per line to allow.
-S, --skip-string-normalization     Don't normalize string quotes or prefixes.
-C, --skip-magic-trailing-comma     Don't use trailing commas as a reason to split lines.
```
The first one is set to 110 (Ensembl guidelines), and I do not see much effect on the second. Will play with the third to see exactly what it means.

File: scripts/brc4/gff-overlaps.py (hunk 2 above)
Comment (JAlvarezJarreta) [reply scripts/brc4/gff-overlaps.py:150]: Agree. These things are good to be taken into account with the coming move to Nextflow, as it takes little effort to revisit the code being migrated for a quick check.

**Comment by JAlvarezJarreta** [comment]
I have tested the `-C` option mentioned before and it seems it does not change the outcome of `argparse`, which matches with what is explained [here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#how-black-wraps-lines). The only option seems to be to add `# fmt: skip` at the end of the lines you want `black` to avoid formatting (this is standard practice with other tools like `pylint` and `mypy` that we plan to introduce soon).

**Comment by JAlvarezJarreta** [comment]
I will apply this change altogether with the tree folder restructuring (PR#49).

**Pull request closed without merging**